### PR TITLE
cmd,interfaces: replace local helpers with cmd.InternalToolPath

### PR DESCRIPTION
**Pull request by bboozzoo**
In the context of #6485, the current code has 2-3 ways for locating internal snapd tools (such as snap-seccomp, snap-update-ns and such). Try to make this consistent and use a single helper.

### CODE REVIEW

File: cmd/cmd_linux.go [hunk 1]
```diff
@@ -115,37 +115,38 @@ func coreSupportsReExec(corePath string) bool {
...
-		logger.Noticef("cannot read /proc/self/exe: %v, using tool outside core", err)
-		return distroTool
+		return "", err
 	}
 
-	// ensure we never use this helper from anything but
-	if !strings.HasSuffix(exe, "/snapd") && !strings.HasSuffix(exe, ".test") {
-		log.Panicf("InternalToolPath can only be used from snapd, got: %s", exe)
-	}
+	if !strings.HasPrefix(exe, dirs.DistroLibExecDir) {
```
Comment (mvo5) [inline cmd/cmd_linux.go]: What will happen if I run `snapd` inside my git tree for testing? Will it no longer find the helpers in /usr/lib/snapd ?

Comment (pedronis) [reply cmd/cmd_linux.go]: easiest is to write a test for the  /home/dev/snapd/snapd case ?

Comment (bboozzoo) [reply cmd/cmd_linux.go]: We do some magic around `/usr/`, so `/home/dev/snapd/snapd` will work. Added a test nonetheless.

**Comment by bboozzoo** [comment]
```
*** Setting version to '2.37.4+git1038.g7f4ff404' from git.

travis_time:end:1084d2e3:start=1552636505705659000,finish=1552636505790687000,duration=85028000
[0Ktravis_fold:end:before_script.1
[0Ktravis_fold:start:before_script.2
[0Ktravis_time:start:25c4cd20
[0K$ go build -o /tmp/snp ./cmd/snap

# github.com/snapcore/snapd/interfaces

interfaces/system_key.go:111:20: undefined: cmd.InternalToolPath

travis_time:end:25c4cd20:start=1552636505810562000,finish=1552636513439435000,duration=7628873000
[0K[31;1mThe command "go build -o /tmp/snp ./cmd/snap" failed and exited with 2 during .[0m
```
Hmmm, but it does work locally.

**Pull request merged**